tqdm/_tqdm.py

```diff
@@ -175,6 +175,26 @@ def tqdm(iterable, desc=None, total=None, leave=False, file=sys.stderr,
         except (TypeError, AttributeError):
             total = None
 
+    if (ncols is None) and (file in (sys.stderr, sys.stdout)):
+        try:
+            from termios import TIOCGWINSZ
+            from fcntl import ioctl
+            from array import array
+        except ImportError:
+            pass
+        else:
+            try:
+                ncols = array('h', ioctl(file, TIOCGWINSZ, '\0' * 8))[1]
+            except SystemExit:
+                raise
+            except:
+                try:
+                    from os import environ
+                except ImportError:
+                    pass
+                else:
+                    ncols = int(environ.get('COLUMNS', 1)) - 1
+
```

Comment:
Hum I don't like the idea to put these lines here. This code is very specific and does one thing : set `ncols`. The best would to put this in a separate file (`utils.py` for example). And then because it's very hard to test, I propose we remove this function from coverage. What do you think ?